### ISSUE TITLE
Add option to disable AUTOPLAY

### ISSUE THREAD

**Issue by ChristianWaits** [recloudstream/cloudstream#1315]
### Describe your suggested feature

In other apps like Stremio, when you click on a certain movie, you are first presented a screen where all your installed addons show all fetched links for that movie. So you can navigate through them and select the one of your preference (maybe a torrent or an online stream, 1080p or 4k, etc). Once you click on a particular link, the player opens and the movie begins.

In Cloudstream things are a little different. First you have to choose the plugin and you open a movie in that particular plugin. But you don't see the different links that plugin fetches until the player is open and the movie already started (or at least is trying to start). That causes problems: Some plugins fetch 20 or more links, and 20 subtitles as well. That takes a considerable amount of time (30 secs-1 min.) until ALL the links and subtitles are fetched. But at the same time, the player is trying to play the first link, which many times is a 4k link, stressing the app and causing frequent crashings. I state that I'm using Cloudstream ONLY on Android TV (on a Philips TV). As you know, Tvs have less ram memory and CPU capacity than phones and are much likely to crash when they are force to do several tasks at the same time. To make things worse, many times the first link returns an error of "time out" and the player AUTOMATICALLY tries to start playing the second one. But on occasions the first link in the background finally finish to load and the audio starts, without the image (as the player is already trying to play another link). The only way to exit from that is stopping app completely, delete cache an restart.

All of that could be easily avoided if there was an OPTION in Settings-Player to turn AUTOPLAY ON or OFF at user preference.
So in my case and probably for all those who run Cloudstream in non-powerful devices, the best choice would be turn autoplay off. So when I click "Play" on a particular movie, the following should happen: The player launches and the plugin starts to load all the streams and subtitles it can fetch, WITHOUT TRYING TO START PLAYING ANYTHING. I gladly wait 30 secs. for that process to finish, and then clicking on "Sources" I could navigate through all the available links and choose the one of my preference, then on the right column I select the subtitle I want, and finally click on "Apply" and only then the movie starts to play.

That autoplay on-off feature would be a simple but huge improvement, at least for TV version.

### Acknowledgements

- [X] My suggestion is **NOT** about adding a new provider
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

**Comment by CranberrySoup** [comment]
You can order how sources are prioritized and loaded. 
If 4k crashes your TV then just make that the lowest priority and it will never auto load 4k videos (unless there only are 4k videos). This system also lets you auto play once a certain quality or source is loaded.

Click the gear button next to "Profile 1" in the sources menu.

**Comment by ChristianWaits** [comment]
I understand. And I know how the Profile configuration works. I have it already well configurated. But depending on the plugin, that settings not always take effect. Even the "Skip loading" popup is pretty useless, since even if you click on it, some plugins keep loading all the links and subtitles anyway. 4k doesn't crash my Tv, in fact I use Cloudstream to play 4k whenever it's available. What crashes TV is the player trying to play a link and fail, then automatically it goes to a second link, that may also fail and at the same time the plugin keeps loading several other links and subtitles. On occasions the first link finally starts when the second one is trying to play and that is a complete mess. That's when Tv crashes. Even if it didn't crash and everything went well and a movie finally started playing, 90% of times it's not the link I would have chosen if I could have seen all the list first. Not to mention subtitles. I have it well configured also to choose subtitles in my language in the first order, but some plugins don't pay attention to that and autoselect a persian subtitle or something like that.

So, even if the autoplay worked, I have to pause, go to sources and re-select both stream and subtitle. So the whole autoplay thing was just a pain. As I said it would be best if the plugin could load all its sources, without trying to play anything. And when loading sources process ends, the users could select the link and subtitle of their preference and finally click play for the movie to begin. I'm not saying that the autoplay disabled should be mandatory, those who prefer things like they are now, won't have to modify anything. I'm just asking for AN OPTION to set autoplay on or off at user preference. What would be the problem to add such an option that would give users more configuration power?

**Comment by CodeCubeNeo** [comment]
Yes I absolutely agree. For example i stared watching something, but then stopped and watched that thing somewhere else and then i want a different video, but the autoplay starts at the old one. I think this is a must have. Please consider this

**Comment by CodeCubeNeo** [comment]
Is there any updates on this?
